src/IECoreHoudini/GEO_CortexPrimitive.cpp

```diff
@@ -113,87 +108,79 @@ GA_Offset GEO_CortexPrimitive::getVertexOffset( GA_Size index ) const
 {
 	return ( index == 0 ) ? m_offset : GA_INVALID_OFFSET;
 }
+#endif
 
 GA_Primitive::GA_DereferenceStatus GEO_CortexPrimitive::dereferencePoint( GA_Offset point, bool dry_run )
 {
-	if ( isDegenerate() )
+	// can dereference the point because there's more than one of these, since the cortex geometry is all based on one vertex, and one point
+	if ( getPointOffset(0) == point )
 	{
-		return GA_DEREFERENCE_DEGENERATE;
+		return isDegenerate() ? GA_DEREFERENCE_DEGENERATE : GA_DEREFERENCE_FAIL;
 	}
-	
-	return GA_DEREFERENCE_FAIL;
+
+	return GA_DEREFERENCE_OK;
 }
 
 GA_Primitive::GA_DereferenceStatus GEO_CortexPrimitive::dereferencePoints( const GA_RangeMemberQuery &pt_q, bool dry_run )
 {
-	if ( isDegenerate() )
+	// if the only point is removed, the primitive is degenerate
+	if ( pt_q.contains( getPointOffset( 0 ) ) )
 	{
-		return GA_DEREFERENCE_DEGENERATE;
+		return GA_DEREFERENCE_DESTROY;
 	}
-	
-	return GA_DEREFERENCE_FAIL;
+	if ( isDegenerate() )
+		return GA_DEREFERENCE_DEGENERATE;
+
+	return GA_DEREFERENCE_OK;
 }
 
 #if UT_MAJOR_VERSION_INT >= 13
 
 void GEO_CortexPrimitive::stashed( bool beingstashed, GA_Offset offset )
 {
 	GEO_Primitive::stashed( beingstashed, offset );
-	
-	if ( beingstashed )
+	// remove reference to the m_object when being stashed
+	if( beingstashed )
 	{
 		m_object = 0;
-		m_offset = GA_INVALID_OFFSET;
 	}
-	else
-	{
-		m_object = 0;
-		m_offset = allocateVertex();
-	}	
 }
 
 #endif
 
 void GEO_CortexPrimitive::stashed( int onoff, GA_Offset offset )
 {
 	GEO_Primitive::stashed( onoff, offset );
-	
-	if ( onoff )
-	{
-		m_object = 0;
-		m_offset = GA_INVALID_OFFSET;
-	}
-	else
-	{
-		m_object = 0;
-		m_offset = allocateVertex();
-	}
+	m_object = 0;
 }
 
 void GEO_CortexPrimitive::clearForDeletion()
 {
 	m_object = 0;
-	m_offset = GA_INVALID_OFFSET;
 	GEO_Primitive::clearForDeletion();
 }
 
 bool GEO_CortexPrimitive::isDegenerate() const
 {
-	return false;
+	// if there's more than one point, then the primitive is degenerate
+	GA_Range points = this->getPointRange();
+	if( points.getEntries() == 1 )
```

Comment:
dereference_ok means that the point can be removed, and the primitive will be unaffected. dereference_destroy means that the if the point has been removed, then the primitive ought to be destroyed, dereference_fail means that it's not possible to remove the point, for some reason, and dereference_degenerate means that if this point is removed, the resulting primitive is no longer a proper primitive of the type it is. I added this because I was getting the occasional crash and I thought maybe it was related to the points being dereferenced, but the primitive not being removed. I don't know for sure though. It also allows points in the primitive to be deleted in 15 and 16.